html/controls/operator.js

```diff
@@ -1089,6 +1088,212 @@ function createJamDialog() {
+		.append($('<option>').attr('value', '1.false').text('Team TO left'))
+		.append($('<option>').attr('value', '2.false').text('Team TO right'))
+		.append($('<option>').attr('value', '1.true').text('Off. Review left'))
+		.append($('<option>').attr('value', '2.true').text('Off. Review right'));
+	
+	var dialog = $("<div>").addClass("NumberDialog");
+	var table = $("<table>").appendTo(dialog);
+	var headers = $("<tr><td/><td/><td/><td/><td/><td/></tr>").appendTo(table);
+	$("<a>").text("Period").addClass("Title")
+		.appendTo(headers.find("td:eq(0)").addClass("Title"));
+	$("<a>").text("After Jam").addClass("Title")
+		.appendTo(headers.find("td:eq(1)").addClass("Title"));
+	$("<a>").text("Duration").addClass("Title")
+		.appendTo(headers.find("td:eq(2)").addClass("Title"));
+	$("<a>").text("Type").addClass("Title")
+		.appendTo(headers.find("td:eq(3)").addClass("Title"));
+	
+	var footer = $("<tr><td/><td/><td/><td/><td/><td/></tr>").attr('id', 'toFooter').appendTo(table);
+	periodDropdownTemplate.clone().appendTo(footer.find('td:eq(0)'));
+	$('<button>').text('Add Timeout').button().click(function() {
+		WS.Set('ScoreBoard.Period('+footer.find('#PeriodDropdown').val()+').InsertTimeout', true);
+	}).appendTo(footer.find('td:eq(3)'));
+	
+	WS.Register(['ScoreBoard.CurrentPeriodNumber'], function(k, v) {
+		var p = Number(v);
+		var i;
+		for (i = lastPeriodRegistered + 1; i <= p; i++) {
+			firstJamListed[p] = 0;
+			lastJamListed[p] = 0;
+			jamDropdownTemplate[i] = $('<select>').attr('id', 'JamDropdown').attr('period', i);
+			WS.Register(['ScoreBoard.Period('+i+').FirstJamNumber'], processFirstJamNumber);
+			WS.Register(['ScoreBoard.Period('+i+').CurrentJamNumber'], processCurrentJamNumber);
+			WS.Register(['ScoreBoard.Period('+i+').Timeout'], processTimeout);
+			lastPeriodRegistered = i;
+		}
+		for (i = lastPeriodListed + 1; i <= p; i++) {
+			var option = $('<option>').attr('value', i).text('P'+i);
+			periodDropdownTemplate.append(option.clone());
+			table.find('#PeriodDropdown').append(option);
+			lastPeriodListed = i;
+		}
+		for (i = lastPeriodListed; i > p; i--) {
+			periodDropdownTemplate.find('option[value='+i+']').remove();
+			table.find('#PeriodDropdown option[value='+i+']').remove();
+			lastPeriodListed = i-1;
+		}
+		footer.find('#PeriodDropdown').val(p);
+	});
+
+	function processFirstJamNumber(k, v) {
+		var p = Number(k.Period);
+		var j = Number(v);
+		var i;
+		if (j == 0 && p > 0) {
+			jamDropdownTemplate[p].find('option').remove();
+			table.find('#JamDropdown[period='+p+'] option').remove();
+			firstJamListed[p] = 0;
+			lastJamListed[p] = 0;
+			return;
+		}
+		if (firstJamListed[p] == 0 && p > 0) {
+			var last = Number(WS.state['ScoreBoard.Period('+p+').CurrentJamNumber']);
+			for (i = j; i <= last; i++) {
+				var option = $('<option>').attr('value', i).text('J'+i);
+				jamDropdownTemplate[p].append(option.clone());
+				table.find('#JamDropdown[period='+p+']').append(option);
```

Comment:
You're ending up with duplicates in this list somehow.

P2 is also not showing, and switching period isn't changing the listed jams.

Reply:
I'm still seeing dupes when I flip between P1 and P2